web/webkit/src/main/scala/net/liftweb/http/HtmlNormalizer.scala

```diff
@@ -0,0 +1,291 @@
+
+import util.Helpers._
+
+/**
+ * Represents an HTML attribute for an event handler. Carries the event name and
+ * the JS that should run when that event is triggered as a String.
+ */
+private case class EventAttribute(eventName: String, jsString: String)
+private object EventAttribute {
+  /**
+   * Some elements allow a URL attribute to take a `javascript:(//)`-style
+   * URL instead of setting an `on*` event in order to invoke JS. For example,
+   * you can (and Lift does) set a form's `action` attribute to
+   * `javascript://(some JS)` instead of setting `onsubmit` to `(someJS);
+   * return false`.
+   *
+   * This is a map from those attribute names to the corresponding JS event
+   * that would be used to execute that JS when it isn't run in line.
+   */
+  val eventsByAttributeName =
+    Map(
+      "action" -> "submit",
+      "href" -> "click"
+    )
+
+  object EventForAttribute {
+    def unapply(attributeName: String): Option[String] = {
+      eventsByAttributeName.get(attributeName)
+    }
+  }
+}
+
+/**
+ * Helper class that performs certain Lift-specific normalizations of HTML
+ * represented as `NodeSeq`s:
+ *  - Fixes URLs for `link`, `a`, `form`, and `script` elements to properly
+ *    prepend the container's context path in case these are absolute URLs.
+ *  - Extracts event attributes (replacing them with
+ *    `[[LiftRules.attributeForRemovedEventAttributes]]` if needed), returning
+ *    instead JavaScript that will attach the corresponding event handler to
+ *    that element.
+ *  - Provides for caller-specific additional processing for each node.
+ */
+private[http] final object HtmlNormalizer {
+  // Fix URLs using Req.normalizeHref and extract JS event attributes for
+  // putting into page JS. Returns a triple of:
+  //  - The optional id that was found in this set of attributes.
+  //  - The normalized metadata.
+  //  - A list of extracted `EventAttribute`s.
+  private def normalizeUrlAndExtractEvents(
+    attributeToNormalize: String,
+    attributes: MetaData,
+    contextPath: String,
+    shouldRewriteUrl: Boolean, // whether to apply URLRewrite.rewriteFunc
+    eventAttributes: List[EventAttribute] = Nil
+  ): (Option[String], MetaData, List[EventAttribute]) = {
+    if (attributes == Null) {
+      (None, Null, eventAttributes)
+    } else {
+      // Note: we don't do this tail-recursively because we have to preserve
+      // attribute order!
+      val (id, normalizedRemainingAttributes, remainingEventAttributes) =
+        normalizeUrlAndExtractEvents(
+          attributeToNormalize,
+          attributes.next,
+          contextPath,
+          shouldRewriteUrl,
+          eventAttributes
+        )
+
+      attributes match {
+        case attribute @ UnprefixedAttribute(
+               EventAttribute.EventForAttribute(eventName),
+               attributeValue,
+               remainingAttributes
+             ) if attributeValue.text.startsWith("javascript:") =>
```

Comment:
For sure, and we do. The mechanism for doing that is the [`EventForAttribute` extractor](https://github.com/lift/framework/blob/ajax-event-extraction/web/webkit/src/main/scala/net/liftweb/http/HtmlNormalizer.scala#L35-L39), since in cases where we detect a `javascript:[//]` URL we *also* need to drop that attribute and replace it with the event attribute (e.g. `href="javascript:alert('hello');"` needs to lose the `href` and add an `onclick`).